src/fault.h

```diff
@@ -0,0 +1,71 @@
+#ifndef FAULT_H
+#define FAULT_H
+
+#include "string_id.h"
+#include "json.h"
+
+class fault;
+using fault_id = string_id<fault>;
+
+class Skill;
+using skill_id = string_id<Skill>;
+
+using quality_id = std::string;
+
+using itype_id = std::string;
+
+class fault
```

Comment:
This class could really use some documentation. And not just something like "Load fault from JSON definition", which basically repeats the function name (load) and parameter type (JsonObject).

Also: Why not use the requirements class for the ... requirements. Gives you one interface for loading from JSON and for use in C++, and it's more generic than a list of qualities and items.